lib/jekyll-titles-from-headings/generator.rb

```diff
@@ -2,9 +2,13 @@ module JekyllTitlesFromHeadings
   class Generator < Jekyll::Generator
     attr_accessor :site
 
-    TITLE_REGEX = %r!\A\s*\#{1,3} (.*)\n$!
+    TITLE_REGEX = %r!\A\s*\#{1,3}\s+(.*)\n$!
```

Comment:
I believe we can't require whitespace after the `#` in the heading. The original implementation of GFM didn't require spaces, and users that move Markdown files that render on GitHub.com to GitHub Pages expect the more lenient syntax to be respected.